go/vt/vtorc/logic/topology_recovery.go

```diff
@@ -551,6 +551,12 @@ func executeCheckAndRecoverFunction(analysisEntry *inst.ReplicationAnalysis) (er
 		return err
 	}
 
+	// Prioritise primary recovery.
+	// If we are performing some other action, first ensure that it is not because of primary issues.
+	if err = recheckPrimaryHealth(analysisEntry, DiscoverInstance, checkIfAlreadyFixed); err != nil {
+		return err
+	}
+
 	// We lock the shard here and then refresh the tablets information
```

Comment:
@bantyK/everyone: hypothetically, what would happen if there was a reparent between `recheckPrimaryHealth` and the `.LockShard`? I suspect this ok, but it would be good to understand

Reply:
This should be ok because vtorc performs a primary health check in the `forceRefreshAllTabletsInShard` method after the locking step like it used to be.